extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQFaultsTest.java

```diff
@@ -132,24 +135,51 @@ public void testInsertTimeOutOfBoundsFault()
   @Test
   public void testInsertTimeNullFault()
   {
+    final String expectedDataSource = "foo1";
+
     final RowSignature rowSignature =
         RowSignature.builder()
                     .add("__time", ColumnType.LONG)
-                    .add("dim1", ColumnType.STRING)
+                    .add("cnt", ColumnType.STRING)
                     .build();
 
-    testIngestQuery()
-        .setSql(
-            "INSERT INTO foo1\n"
-            + "SELECT TIME_PARSE(dim1) AS __time, dim1 as cnt\n"
-            + "FROM foo\n"
-            + "PARTITIONED BY DAY\n"
-            + "CLUSTERED BY dim1")
-        .setExpectedDataSource("foo1")
-        .setExpectedRowSignature(rowSignature)
-        .setExpectedSegment(ImmutableSet.of(SegmentId.of("foo", Intervals.of("2000-01-01T/P1M"), "test", 0)))
-        .setExpectedMSQFault(InsertTimeNullFault.instance())
-        .verifyResults();
+    final String sql = "INSERT INTO foo1\n"
+                     + "SELECT TIME_PARSE(dim1) AS __time, dim1 as cnt\n"
+                     + "FROM foo\n"
+                     + "PARTITIONED BY DAY\n"
+                     + "CLUSTERED BY dim1";
```

Comment:
If you were to add a `WHERE TIME_PARSE(dim1) is not null` to the query, do the results of the two queries becomes the same regardless of mode?  I think they should, but curious.

Reply:
Yes, in that case, the results are the same: both queries ingest the two rows with parseable timestamps, and ignore the four with unparseable timestamps.